dali/pipeline/operator/builtin/external_source.h

```diff
@@ -335,15 +334,15 @@ class ExternalSource : public Operator<Backend> {
   bool blocking_ = true;
   bool no_copy_ = false;
 
-  // now it only indicates that there is a data in the ExternalSource, in the future
-  // a per sample metadata could be stored there
-  struct ExtenralSourceState {
-    bool dummy;
+  // now it only indicates that there is data in the ExternalSource, in the future
+  // a per sample metadata could be stored here
+  struct ExternalSourceState  {
+    bool copied_shared_data = false;
   };
 
-  std::list<ExtenralSourceState> state_;
+  std::list<ExternalSourceState > state_;
 
-  bool gpu_noncontinuous_zero_copy = false;
+  bool gpu_noncontiguous_zero_copy = false;
```

Comment:
Small nitpick:
```suggestion
  bool gpu_noncontiguous_zero_copy_ = false;
```
Also I find the name the opposite of what it actually did, maybe add a comment here?

Reply:
Done